src/truncate.jl

```diff
@@ -95,6 +95,9 @@ struct Truncated{D<:UnivariateDistribution, S<:ValueSupport, T<: Real, TL<:Union
     logtp::T      # log(tp), i.e. log(ucdf - lcdf)
 
     function Truncated(d::UnivariateDistribution, l::TL, u::TU, loglcdf::T, lcdf::T, ucdf::T, tp::T, logtp::T) where {T <: Real, TL <: Union{T,Nothing}, TU <: Union{T,Nothing}}
+        if iszero(tp) && !(l == u && insupport(d, l))
```

Comment:
Can we check `logtp` instead? Seems safer if the untruncated has tiny non-zero mass on the truncated interval. Also, I think the second condition is not needed since `tp = P(lower <= X <= upper)` with supposedly correct handling of discrete and degenerate distributions (but maybe it's not computed correctly in some cases currently?).
```suggestion
        if logtp == -Inf
```